ospf6d/ospf6_route.c

```diff
@@ -1058,34 +1066,80 @@ void ospf6_route_show(struct vty *vty, struct ospf6_route *route)
 	else
 		prefix2str(&route->prefix, destination, sizeof(destination));
 
-	i = 0;
-	for (ALL_LIST_ELEMENTS_RO(route->nh_list, node, nh)) {
-		struct interface *ifp;
-		/* nexthop */
-		inet_ntop(AF_INET6, &nh->address, nexthop, sizeof(nexthop));
-		ifp = if_lookup_by_index_all_vrf(nh->ifindex);
-		if (!i) {
-			vty_out(vty, "%c%1s %2s %-30s %-25s %6.*s %s\n",
-				(ospf6_route_is_best(route) ? '*' : ' '),
-				OSPF6_DEST_TYPE_SUBSTR(route->type),
-				OSPF6_PATH_TYPE_SUBSTR(route->path.type),
-				destination, nexthop, IFNAMSIZ, ifp->name,
-				duration);
-			i++;
-		} else
-			vty_out(vty, "%c%1s %2s %-30s %-25s %6.*s %s\n", ' ',
-				"", "", "", nexthop, IFNAMSIZ, ifp->name, "");
+	if (use_json) {
+		json_route = json_object_new_object();
+		json_object_string_add(json_route, "destination", destination);
+		json_object_boolean_add(json_route, "isBestRoute",
+					ospf6_route_is_best(route));
+		json_object_string_add(json_route, "destinationType",
+				       OSPF6_DEST_TYPE_SUBSTR(route->type));
+		json_object_string_add(
+			json_route, "pathType",
+			OSPF6_PATH_TYPE_SUBSTR(route->path.type));
+		json_object_string_add(json_route, "duration", duration);
+	}
+
+	/* Nexthops */
+	if (use_json) {
+		json_array_next_hops = json_object_new_array();
+		for (ALL_LIST_ELEMENTS_RO(route->nh_list, node, nh)) {
```

Comment:
Let's move the loop to outside so we have it 1 time and put the use_json code on the inside